src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -977,7 +973,8 @@ private bool ConsumeStringNextSegment()
                 _bytePositionInLine += localBuffer.Length;
             }
 
-            SequencePosition start = new SequencePosition(startPosition.GetObject(), startPosition.GetInteger() + startConsumed);
+            SequencePosition originalPosition = rollBackState._prevCurrentPosition;
+            SequencePosition start = new SequencePosition(originalPosition.GetObject(), originalPosition.GetInteger() + rollBackState._prevConsumed + 1);
```

Comment:
Is this logic correct?  If the last thing in segment0 is the start quote is the value `(segment0, segment0.Length)` or is it `(segment1, 0)`?  Does SequencePosition handle holding segment.Length?

Reply:
I believe either works and isn't really a correctness issue (the sequence we return ends up having an empty segment up front). This logic is unchanged within this PR (and the issue exists in master), so I am going to leave this as is for now.